// ng-qrcode/ng-qrcode.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var platform_browser_1 = require("@angular/platform-browser");
var QRCodeComponent = (function () {
    function QRCodeComponent(sanitizer) {
        this.sanitizer = sanitizer;
    }
    QRCodeComponent.prototype.ngOnInit = function () {
        this.src = this.sanitizer.bypassSecurityTrustResourceUrl('https://api.qrserver.com/v1/create-qr-code/?data='
            + this.data
            + '&size='
            + this.size
            + 'x'
            + this.size
            + '&margin=0');
    };
    return QRCodeComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], QRCodeComponent.prototype, "data", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Number)
], QRCodeComponent.prototype, "size", void 0);
QRCodeComponent = __decorate([
    core_1.Component({
        selector: 'qr-code',
        template: "\n    <img [src]=\"src\" [alt]=\"data\" width=\"{{size}}px\" height=\"{{size}}px\">\n  "
    }),
    __metadata("design:paramtypes", [platform_browser_1.DomSanitizer])
], QRCodeComponent);
exports.QRCodeComponent = QRCodeComponent;
var QRCodeModule = (function () {
    function QRCodeModule() {
    }
    return QRCodeModule;
}());
QRCodeModule = __decorate([
    core_1.NgModule({
        imports: [common_1.CommonModule],
        declarations: [QRCodeComponent]
    })
], QRCodeModule);
exports.QRCodeModule = QRCodeModule;
